src/main/java/org/tron/core/Blockchain.java

```diff
@@ -57,6 +59,8 @@
     private byte[] lastHash;
     private byte[] currentHash;
 
+    private Client client;
+
     /**
      * create new blockchain
      *
```

Comment:
I'm wondering if it is possible to inject the database when initializing the Blockchain for the first time from this constructor:
`public Blockchain(String address, String type) {`

Doing so would avoid making the Blockchain responsible for setting the database connection, which to me isn't an intuitive responsibility of this class.

Doing so would mean the logic for deciding whether to load an existing or create a new blockchain would have to change to some other way.

I wonder if `if(blockDB.getData(LAST_HASH) != null) { ... }` is sufficient for this purpose?

Reply:
Initialize the database when the program starts. 

Solution A:
```java
// Initialize on the outside
DB blockDB = new DB(); 
// Incoming through formal parameters
Blockchain blockchain = new Blockchain(blockDB); 
```